rasa/core/agent.py

```diff
@@ -400,6 +400,30 @@ def is_ready(self):
             and self.interpreter is not None
         )
 
+    async def parse_message_from_text(self, message_data: Text):
+        """Handles message text and intent payload input messages.
+
+        The return value of this function is parsed_data.
+
+        Example:
+
+        By sending /greet{"name":"Rasa"}:
+
+        parsed_data = {
+            "text": '/greet{"name":"Rasa"}',
+            "intent": {"name": "greet", "confidence": 1.0},
+            "intent_ranking": [{"name": "greet", "confidence": 1.0}],
+            "entities": [{"entity": "name", "start": 6, "end": 21, "value": "Rasa"}],
+        }
+        """
+        try:
+            processor = self.create_processor()
+            message = UserMessage(message_data)
+            parsed_data = await processor._parse_message(message)
+            return parsed_data
+        except Exception:
+            logger.exception("Failed to parse message data from text.")
```

Comment:
I think I would rather handle this with a status code in the server, e.g. return a `400` in case the the parsing failed.

Reply:
Do you have any suggestion on how to do it?

In `server.py` there is a class called `ErrorResponse(Exception)` it is possible to raise status exceptions like this:
```
except Exception as e:                                                                                  
    logger.debug(traceback.format_exc())                                                                
    raise ErrorResponse(                                                                                
        500, "ParsingError", "An unexpected error occurred. Error: {}".format(e)                        
    )
```
But this class isn't at `agent.py`. I'm not sure if I need to import this on `agent.py` and then `raise` the `400` code or simply  `raise`

Reply:
You should catch the exception here, but catch it in the server method. Then you can use the snippet you just mentioned.

Reply:
Hmm, so remove the `try` from this method and add it to `server.py`?

Reply:
Yes. Would be my suggestion.

Reply:
Thanks :smile: